### DISCUSSION TITLE
Winlogbeat SSL encryption

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#4652] [Unsupported Versions] [answered]
Hello!

I followed the instructions in the SO documentation and copied my relevant certs and key to "/opt/so/conf/logstash/etc/certs/", changed the configuration file "0009_input_beats.conf" to enable SSL encryption as explained in the SO documentation, but when I restart logstash on my manager node it only removes all files in "/opt/so/conf/logstash/etc/certs/" and throws the following error in my logstash.log:

input {
    beats {
      # This setting must be a path
      # File does not exist or cannot be opened /usr/share/logstash/certs/my-manager.crt
      ssl_certificate => "/usr/share/logstash/certs/my-manager.crt"
      ...
    }
  }

[2021-06-30T14:24:15,485][ERROR][logstash.inputs.beats    ] Invalid setting for beats input plugin:

  input {
    beats {
      # This setting must be a path
      # File does not exist or cannot be opened /usr/share/logstash/certs/my-manager.key
      ssl_key => "/usr/share/logstash/certs/my-manager.key"
      ...
    }
  }

What is missing or not correct?

Cheers, Ben

**Comment by ben-sec** [comment, marked as answer]
I found the solution in [this discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/2503). You have to put the certs and key in the following directory: /opt/so/saltstack/local/salt/logstash/etc/certs

Then run "so-logstash-restart" and the files will be placed permanently in "/opt/so/conf/logstash/etc/certs" and can be accessed in the docker container of logstash via "/usr/share/logstash/certs".

The SO team should update the [beats documentation](https://docs.securityonion.net/en/2.3/beats.html?highlight=beat#encryption) accordingly. ;-)

**Reply by jaredatobe** [reply]
This has been updated in documentation as you requested. See the following discussion:

https://github.com/Security-Onion-Solutions/securityonion/discussions/4964
